lib/trino-hive-formats/src/test/java/io/trino/hive/formats/line/openxjson/TestOpenxJsonFormat.java

```diff
@@ -884,14 +884,35 @@ public void testDate()
         assertDate("\"1970-01-02\"", 1);
         assertDate("\"1969-12-31\"", -1);
 
-        // Hive ignores everything after the first space
-        assertDate("\"1986-01-01 anything is allowed here\"", LocalDate.of(1986, 1, 1).toEpochDay());
-
-        assertDate("\"1986-01-01\"", LocalDate.of(1986, 1, 1).toEpochDay());
-        assertDate("\"1986-01-33\"", LocalDate.of(1986, 2, 2).toEpochDay());
-
-        assertDate("\"5881580-07-11\"", Integer.MAX_VALUE);
-        assertDate("\"-5877641-06-23\"", Integer.MIN_VALUE);
+        // Single digit month and day
+        assertDate("\"2025-4-2 12341234\"", LocalDate.of(2025, 4, 2).toEpochDay());
+        assertDate("\"2025-09-1 12341234\"", LocalDate.of(2025, 9, 1).toEpochDay());
+        assertDate("\"2025-8-01 12341234\"", LocalDate.of(2025, 8, 1).toEpochDay());
+
+        // Hive ignores everything after yyyy-mm-dd
+        long day = LocalDate.of(1986, 1, 1).toEpochDay();
+        SqlDate sqlDate = new SqlDate((int) day);
+        assertDate("\"1986-01-01 anything is allowed here\"", day);
+        assertValueTrinoOnly(DATE, "\"1986-01-01T00:00:00.000Z\"", sqlDate);
+        assertValueTrinoOnly(DATE, "\"1986-01-01AA00:00:00.000Z\"", sqlDate);
+
+        // Test bad yyyy-mm-dd formats
+        // Cascading when number of digits are allowed
+        assertValueTrinoOnly(DATE, "\"1986-01-33\"", null);
+        assertValueTrinoOnly(DATE, "\"2023-80-80\"", null);
+        assertValueTrinoOnly(DATE, "\"9999-99-99\"", null);
+
+        // Digits are more than allowed in year, month, or day
+        assertValueTrinoOnly(DATE, "\"02025-01-01\"", null);
+        assertValueTrinoOnly(DATE, "\"2025-011-01\"", null);
+        assertValueTrinoOnly(DATE, "\"2025-01-011\"", null);
+
+        // Digits are more than allowed in year, month, or day and cascading
+        assertValueTrinoOnly(DATE, "\"202510-01-01\"", null);
+        assertValueTrinoOnly(DATE, "\"2025-100-01\"", null);
+        assertValueTrinoOnly(DATE, "\"2025-01-100\"", null);
+        assertValueTrinoOnly(DATE, "\"5881580-07-11\"", null);
+        assertValueTrinoOnly(DATE, "\"-5877641-06-23\"", null);
 
```

Comment:
Why are all of these "Trino Only" tests?

Reply:
I think the issue is that the "Hive" implementation uses the starburst OpenX JsonSerDe which uses Hive's date parsing, whereas the original OpenX SerDe actually uses `SimpleDateFormat` which implies different semantics in the edge cases being addressed here. Correct me if I'm wrong on that @ljw9111